src/openrct2/actions/MapChangeSizeAction.cpp

```diff
@@ -63,6 +70,9 @@ GameActions::Result MapChangeSizeAction::Execute() const
         MapExtendBoundarySurfaceY();
     }
 
+    // Shift the map (allows increasing the map at the 0,0 position
+    ShiftMap(_shift);
```

Comment:
Note: check if it correctly handles shifting a (say) 999×999 map by 1×1, or a 995×995 map by 8×8 (which you cannot do).